### PR TITLE
Framework: Run create-class codemod

### PR DESCRIPTION
**Pull request by ockham**
WIP. 

```
client/layout/nux-welcome/welcome-message.jsx: `` was skipped because of invalid field(s) `analyticsEvents` on the React component. Remove any right-hand-side expressions that are not simple, like: `componentWillUpdate: createWillUpdate()` or `render: foo ? renderA : renderB`.
client/blocks/author-selector/index.jsx: `` was skipped because of deprecated API calls. Remove calls to getDOMNode, isMounted, replaceProps, replaceState, setProps in your React component and re-run this script.
client/blocks/term-tree-selector/terms.jsx: `TermTreeSelectorList` was skipped because of API calls that will be removed. Remove calls to `getDefaultProps` and/or `getInitialState` in your React component and re-run this script.
client/post-editor/post-editor.jsx: `PostEditor` was skipped because of API calls that will be removed. Remove calls to `getDefaultProps` and/or `getInitialState` in your React component and re-run this script.
client/layout/index.jsx: `Layout` was skipped because of inconvertible mixins.
client/components/infinite-list/index.jsx: `` was skipped because of deprecated API calls. Remove calls to getDOMNode, isMounted, replaceProps, replaceState, setProps in your React component and re-run this script.
client/blocks/credit-card-form/index.jsx: `CreditCardForm` was skipped because of invalid field(s) `fieldNames` on the React component. Remove any right-hand-side expressions that are not simple, like: `componentWillUpdate: createWillUpdate()` or `render: foo ? renderA : renderB`.
```

```
client/components/global-notices/index.jsx: `NoticesList` was skipped because of inconvertible mixins.
client/components/happychat/timeline.jsx: `Timeline` was skipped because of inconvertible mixins.
client/components/happychat/composer.jsx: `Composer` was skipped because of inconvertible mixins.
client/components/infinite-list/index.jsx: `` was skipped because of deprecated API calls. Remove calls to getDOMNode, isMounted, replaceProps, replaceState, setProps in your React component and re-run this script.
client/components/stat-update-indicator/index.jsx: `StatUpdateIndicator` was skipped because of deprecated API calls. Remove calls to getDOMNode, isMounted, replaceProps, replaceState, setProps in your React component and re-run this script.
```

### CODE REVIEW

**Comment by ockham** [comment]
Looks like the codemod ignores instances of `export default localize( React.createClass( /*...*/ ) )`. Looks like it's time for your #18710 @spen 😄

**Comment by samouri** [comment]
also dependent on deprecated api call removal like [isMounted](https://github.com/Automattic/wp-calypso/issues/18926)

**Comment by samouri** [comment]
I've commandeered this pull request :)

**Comment by samouri** [comment]
check this out: 

```bash
git grep 'React.createClass' | grep -v "README"
bin/codemods/src/assign-component-to-displayname-variable.js: export default someHoC( React.createClass( {
bin/codemods/src/assign-component-to-displayname-variable.js: const SomeDisplayName = React.createClass({
bin/codemods/src/assign-component-to-displayname-variable.js:	// can work on both es6 classes and React.createClass functions.
bin/codemods/src/assign-component-to-displayname-variable.js:				? // if createClass, replace with const x = React.createClass...
bin/codemods/src/i18n-mixin-to-localize.js:	// Find the declaration to wrap with the localize HOC. It can be the React.createClass
docs/coding-guidelines/html.md:const Post = React.createClass({
docs/coding-guidelines/html.md:const Post = React.createClass({
docs/coding-guidelines/javascript.md:- Use [stateless function components or the `React.Component` class](https://facebook.github.io/react/docs/components-and-props.html#functional-and-class-components) instead of `React.createClass`
docs/coding-guidelines/javascript.md:  - Unlike `React.createClass`, methods of components extending `React.Component` are not automatically bound to the instance. Instead, you will need to bind the functions in your component's constructor or use [class instance property initializers](https://github.com/tc39/proposal-class-public-fields)
docs/testing/testing-overview.md:	See [#5221](https://github.com/Automattic/wp-calypso/pull/5221) and [#18064](https://github.com/Automattic/wp-calypso/pull/18064) for full examples using `React.createClass()` and ES6 classes, respectively.

```

File: client/components/bulk-select/docs/example.jsx [hunk 1]
```diff
@@ -12,10 +12,14 @@ import React from 'react';
 import Card from 'components/card';
 import BulkSelect from 'components/bulk-select';
 
-export default React.createClass( {
-	displayName: 'BulkSelects',
+export default class extends React.Component {
+    static displayName = 'BulkSelects';
 
-	handleToggleAll( checkedState ) {
+	state = {
```
Comment (samouri) [inline client/components/bulk-select/docs/example.jsx]: so beautiful

**Comment by samouri** [comment]
Note: i had done this successfully with reader in the past: https://github.com/Automattic/wp-calypso/pull/13965

**Comment by bsessions85** [comment]
E2E tests all passed

File: client/components/input-chrono/docs/example.jsx [hunk 2]
```diff
@@ -17,43 +16,40 @@ import Card from 'components/card';
...
-				<Card style={ { width: '300px', margin: 0 } }>
-					<InputChrono value={ this.state.date.calendar() } onSet={ this.onSet } />
-				</Card>
-			);
-		},
-	} )
-);
+const InputChronoExample = localize(/**
+ * Date Picker Demo
+ */
```
Comment (ockham) [inline client/components/input-chrono/docs/example.jsx]: Ugh, we should fix this. I'll push a commit.

Comment (samouri) [reply client/components/input-chrono/docs/example.jsx]: nice catch. it also occured in `client/components/post-schedule/docs/example.jsx`.

**Comment by ockham** [comment]
Amazing job @samouri! :champagne:

**Pull request merged**